changelog/upgrade-notes.md

```diff
@@ -31,6 +31,10 @@ but with additional css:
 }
 ```
 
+#### Removal of `ol.source.TileVector`
+
+The experimental `ol.source.TileVector` class has been removed. Its functionality is now provided by `ol.source.VectorTile`. The API is the same, except for the `tileLoadFunction`, which is no longer an `ol.TileVectorLoadFunctionType`, but an `ol.TileLoadFunctionType`.
+
```

Comment:
Since we just agreed on the removal of compatibility with legacy ol.source.TileVector, I'd say it makes sense to stick with what's in the pull request - ol.VectorTile, ol.source.VectorTile, ol.layer.VectorTile.